frontend/src/toolbar/flags/FlagsToolbarMenu.tsx

```diff
@@ -16,11 +16,18 @@ import { toolbarConfigLogic } from '~/toolbar/toolbarConfigLogic'
 
 export const FlagsToolbarMenu = (): JSX.Element => {
     const { searchTerm, filteredFlags, userFlagsLoading } = useValues(flagsToolbarLogic)
-    const { setSearchTerm, setOverriddenUserFlag, deleteOverriddenUserFlag, getUserFlags, checkLocalOverrides } =
-        useActions(flagsToolbarLogic)
-    const { apiURL } = useValues(toolbarConfigLogic)
+    const {
+        setSearchTerm,
+        setOverriddenUserFlag,
+        deleteOverriddenUserFlag,
+        getUserFlags,
+        checkLocalOverrides,
+        setFeatureFlagValueFromPostHogClient,
+    } = useActions(flagsToolbarLogic)
+    const { apiURL, posthog: posthogClient } = useValues(toolbarConfigLogic)
 
     useEffect(() => {
+        posthogClient?.onFeatureFlags(setFeatureFlagValueFromPostHogClient)
```

Comment:
This posthogClient is actually the user's posthog, not the one we wrap into the toolbar (because that will have our flags, not the users flags 😓 ), and is actually set here: https://github.com/PostHog/posthog-js/blob/main/src/extensions/toolbar.ts#L155 so we'll be ok using any random posthog-js function :)

Reply:
And yes I was very confused for a bit trying to find the right one too 😅

Reply:
Mmmmm I think you're mixing things up - this `posthogClient` is the one passed in from the user's browser (so posthog-js)